dags/analyzer_helper/discourse/extract_raw_members.py

```diff
@@ -0,0 +1,47 @@
+from github.neo4j_storage.neo4j_connection import Neo4jConnection
+
+class ExtractRawMembers:
+    def __init__(self, forum_endpoint:str):
+        """
+        Initialize the ExtractRawMembers with the Neo4j connection parameters.
+        """
+        self.neo4jConnection = Neo4jConnection()
+        self.driver = self.neo4jConnection.connect_neo4j()
+        self.forum_endpoiont = forum_endpoint
+
+    def close(self):
+        """
+        Close the Neo4j connection.
+        """
+        self.driver.close()
+
+    def fetch_member_details(self):
+        """
+        Fetch details of members from the Discourse forum.
+
+        :return: List of dictionaries containing member details.
+        """
+        query = """
+        MATCH (forum:DiscourseForum {endpoint: $forum_endpoint})
+        MATCH (user:DiscourseUser)-[:HAS_JOINED]->(forum)
+        MATCH (user:DiscourseUser)-[:HAS_BADGE]->(badge)
```

Comment:
Two notes on this query

1. Some users might not have the badge. Please do an `OPTIONAL MATCH` for the second MATCH
2. For the second `MATCH` you don't need to get any new users, so Just doing the MATCH operation like this could be right. `MATCH (user) -[:HAS_BADGE]->`

So the first part of the query could become
```
MATCH (forum:DiscourseForum {endpoint: $forum_endpoint})
MATCH (user:DiscourseUser)-[:HAS_JOINED]->(forum)
MATCH (user)-[:HAS_BADGE]->(badge)
```